teps/0008-support-knative-service-for-triggers-eventlistener-pod.md

```diff
@@ -0,0 +1,318 @@
+and if eventlistener pod deployed by Kubernetes Deployment then autoscale needs to be handled by triggers explicitly (`may be HPA`)in order to support max number of request,
+So usage of Knative Service solves the autoscale problem by default(`using KPA`).
+
+Though this is a bit rare case because right now triggers are in `alpha` so not sure about the usage per seconds.
+
+To summarize, Knative does not only promise to scale-out (and let’s be honest, there likely won’t be millions of events), but also scale-to-zero. As Knative(`Serverless Solution`) is pay-per-use which ideally costs based on the usage.
+
+#### Easy Pluggable:
+Knative is built on top of Kubernetes and the yaml looks like Deployment so there won't be difficulty to the user with respect usage.
+Also Knative handles k8s service creation by default with public accessibility which in turn available as part of eventlistener status address.
+
+##### Note:
+Triggers provide flexibility to the user to deploy Knative Service but not the installation of Knative itself and its responsibility of the user to have Knative running beforehand.
+
+### Usage examples
+
+#### Default EventListener yaml
+For Backward compatibility the default behavior will be as it is for few releases.
+
+```yaml
+
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ serviceAccountName: tekton-triggers-example-sa
+ serviceType: NodePort
+ podTemplate:
+   nodeSelector:
+     app: test
+   tolerations:
+   - key: key
+     value: value
+     operator: Equal
+     effect: NoSchedule
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+``` 
+
+#### Kubernetes Based 
+This is exactly the same whatever we have right now with default.
+The reason to move `serviceAccountName`, `podTemplate`, 
+to `typed` field is because those are part of [WithPod{}](https://github.com/knative/pkg/blob/master/apis/duck/v1/podspec_types.go#L41) duck type
+and which helps to support any of the pod and container field without hardcoding in [podTemplate](https://github.com/tektoncd/triggers/blob/master/pkg/apis/triggers/v1alpha1/event_listener_types.go#L62).
+
+1.If user specify podSpec fields.  
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+ name: github-listener-interceptor
+spec:
+ triggers:
+   - name: foo-trig
+     interceptors:
+       - github:
+           secretRef:
+             secretName: foo
+             secretKey: bar
+           eventTypes:
+             - pull_request
+     bindings:
+       - ref: pipeline-binding
+     template:
+       name: pipeline-template
+ service:
```

Comment:
v. minor subjective issue: I wonder if there is a more descriptive name than `typed` and `dynamic`.

Reply:
I used `typed` to indicate Kubernetes Deployment because there we know the Type of `Kind`
where as `dynamic` to indicate it can be any `Kind`

But i understand that these are not that user friendly 

How about these

**typed** to `kubernetesResource` or `deploymentResource` or `builtinResource`
**dynamic** to `customResource`

I am open to the suggestions :smile_cat: :see_no_evil: :stuck_out_tongue:

Reply:
yeah, we can keep discussing without letting this be a blocker :)